File: generator/src/test/java/ch/bailu/gtk/TestAlias.java
package ch.bailu.gtk;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

import ch.bailu.gtk.converter.NamespaceType;
import ch.bailu.gtk.table.AliasTable;

public class TestAlias {

    @Test void testAlias() {
        var from = new NamespaceType("gtk", "Allocation");
        var to = new NamespaceType("gdk", "Rectangle");


        AliasTable.INSTANCE.add(from, to);

        var test = AliasTable.INSTANCE.convert(from);

        assertEquals("Rectangle", test.getName());
        assertEquals("gdk", test.getNamespace());
    }
}

File: generator/src/test/kotlin/ch/bailu/gtk/TestAlias.kt
package ch.bailu.gtk

import ch.bailu.gtk.model.type.NamespaceType
import ch.bailu.gtk.table.AliasTable.add
import ch.bailu.gtk.table.AliasTable.convert
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class TestAlias {
    @Test
    fun testAlias() {
        val from = NamespaceType("gtk", "Allocation")
        val to = NamespaceType("gdk", "Rectangle")
        add(from, to)
        val test = convert(from)
        Assertions.assertEquals("Rectangle", test.name)
        Assertions.assertEquals("gdk", test.namespace)
    }
}
